core/workspace_svg.js

```diff
@@ -1110,6 +1110,17 @@ Blockly.WorkspaceSvg.prototype.isDeleteArea = function(e) {
   return Blockly.DELETE_AREA_NONE;
 };
 
+/**
+ * Is the mouse event outside the blocks UI, to the right of the workspace?
+ * @param {!Event} e Mouse move event.
+ * @return {boolean} True if event is outside the blocks UI.
+ */
+Blockly.WorkspaceSvg.prototype.isOutside = function(e) {
+  var mousePoint = Blockly.utils.mouseToSvg(e, this.getParentSvg(),
+      this.getInverseScreenCTM());
+  return this.getParentSvg().width.baseVal.value < mousePoint.x;
```

Comment:
I'm a bit leery of this implementation, because it assumes that the only "outside" is to the right side.

Note that Blockly assumes that there's nothing useful to the "outside" of the toolbox, based on the configuration of the workspace, so it's not like we're doing the right thing either.

I think this will break in RTL.  If you keep the editor in the same place but inject with RTL, everything you're looking for will be in the delete area.  If the editor and stage swap places, this will just be looking on the wrong side.

Reply:
I think eventually I can have isOutside take an RTL argument and return the right thing for RTL. Is there a way to set to RTL now?

Reply:
Use [this.RTL](https://github.com/LLK/scratch-blocks/blob/develop/core/workspace.js#L47)